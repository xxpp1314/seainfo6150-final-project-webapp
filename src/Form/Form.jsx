import React, { useState } from "react";
import styles from "./Form.module.css";
import { Link } from "react-router-dom";


const Form = () => {
  const [submittedForm, setSubmittedForm] = useState();

  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  };

  return (
    <div>
      <div>
      {
        submittedForm ? (
          <div className={styles.text}>
            <h2 className={styles.text}>Please review your information below</h2><br />
            Your First name: <b className={styles.reviewtext}>{submittedForm.get("myFirstName")}</b>.<br /><br />
            Your Last Name: <b className={styles.reviewtext}>{submittedForm.get("myLastName")}</b>.<br /><br />
            Your email ID: <b className={styles.reviewtext}>{submittedForm.get("myEmailID")}</b>.<br /><br />
            Your Phone Number: <b className={styles.reviewtext}>{submittedForm.get("myPhone")}</b>.<br /><br />
            Home Address Line 1: <b className={styles.reviewtext}>{submittedForm.get("myAddressLine1")}</b>.<br /><br />
            Home Address Line 2: <b className={styles.reviewtext}>{submittedForm.get("myAddressLine2")}</b>.<br /><br />
            City of the Home Address: <b className={styles.reviewtext}>{submittedForm.get("myCity")}</b>.<br /><br />
            State of the Home Address: <b className={styles.reviewtext}>{submittedForm.get("myState")}</b>.<br /><br />
            Your Zip Code: <b className={styles.reviewtext}>{submittedForm.get("myZipCode")}</b>.<br /><br />
            Do you have any experiences with pet? <b className={styles.reviewtext}>{submittedForm.get("myExperience")}</b>.<br /><br />
            What is your home type? <b className={styles.reviewtext}>{submittedForm.get("myHome")}</b>.<br /><br />
            Does your home have any restrctions related to this pet? <b className={styles.reviewtext}>{submittedForm.get("myRestriction")}</b>.<br /><br />
            
            <Link to="/thankyou"> 
            <button className={styles.button} type="button">Submit</button><br /><br />
            </Link>
          </div>
        ) : (
         <div className={styles.formdetails}>
            <form onSubmit={onSubmit} className={styles.text}>
            <br /><label htmlFor="name">Your First Name: </label><br />
            <input  className={styles.space} type="text" name="myFirstName" id="myFirstName" required/><br /><br />

            <label htmlFor="myTextId">Your Last Name:  </label><br />
            <input className={styles.space} type="text" name="myLastName" id="myLastName" required/><br /><br />

            <label htmlFor="myTextId">Your Email ID:  </label><br />
            <input className={styles.space} type="email" name="myEmailID" id="myEmailID" placeholder="abc@example.com" required/><br /><br />

            <label htmlFor="myTextId">Your Phone No: </label><br />
            <input className={styles.space} type="tel" name="myPhone" id="myPhone" pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}" placeholder= "123-345-6785" required/><br /><br />

            <label htmlFor="myTextId">Address Line 1: </label>
            <input  type="text" name="myAddressLine1" id="myAddressLine1" className={styles.details} maxLength="500" required/><br /><br />

            <label htmlFor="myTextId">Address Line 2: </label>
            <input type="text" name="myAddressLine2" id="myAddressLine2" className={styles.details} maxLength="500"/><br /><br />

            <label htmlFor="myTextId">City of the Home Address: </label><br />
            <select className={styles.dropdown} name="myCity" id="myCity">
              <option value="Seattle">Seattle</option>
              <option value="San Jose">San Francisco</option>
              <option value="Boston">Boston</option>
            </select><br /><br /> 

            <label htmlFor="myTextId">State of the Home Address: </label><br />
            <select className={styles.dropdown} name="myState" id="myState">
              <option value="WA">Washington</option>
              <option value="CA">California</option>
              <option value="MA">Massachusetts</option>
            </select><br /><br />

            <label htmlFor="myTextId">Your Zip Code: </label><br />
            <input className={styles.space} type="zip" name="myZipCode" id="myZipCode" pattern="[0-9]{5}" placeholder= "12345" required/><br /><br />

            <label htmlFor="myTextId">Do you have any experiences with pet?  </label>
            <select className={styles.dropdown} name="myExperience" id="myExperience">
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select><br /><br />

            <label htmlFor="myTextId">What is your home type?  </label><br />
            <select className={styles.dropdown} name="myHome" id="myHome">
              <option value="Apartment">Apartment</option>
              <option value="House">Single Family House/Town House</option>
              <option value="Other">Other</option>
            </select><br /><br />

            <label htmlFor="myTextId">Does your home have any restrctions related to this pet? </label><br />
            <select className={styles.dropdown} name="myRestriction" id="myRestriction">
              <option value="Yes">Yes</option>
              <option value="No">No</option>
            </select><br /><br />

            <input className={styles.button} type="submit" value="Submit" /><br /><br />
          </form>
        </div>
        )
      }
    </div>
    </div>
  )
}

export default Form