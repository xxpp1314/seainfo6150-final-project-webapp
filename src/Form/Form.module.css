.formStyle {
    padding: 2%;
    font-size: 100%;
    color:  rgb(243, 222, 222);;
    text-align: center;
    line-height: 2.5%;
    border-radius: 2%;
  }

  .text{
    color:rgb(243, 222, 222);
    font-weight: lighter;
    font-family: 'Lora', serif;
  }

  .reviewtext{
    color: rgb(182, 104, 104);
  }

  .details {
    width: 100%;
    background-color: rgb(243, 222, 222);
  }

  .dropdown{
    height: 100%;
    width: 40%;
    background-color: rgb(243, 222, 222);
  }

  .space{
    padding-right: 22%;
    height: 100%;
    width: 30%;
    background-color: rgb(243, 222, 222);
  }

  .formdetails{
    padding: 5%;
    box-shadow: 0px 0px 5px 2px rgba(204, 0, 0, 0.5);    
    border: 0px solid #666;
    border-radius: 2%;
    align-items: flex-start;
    margin: 1%;
  }

  .button{
    align-self: center;
    color:white;
    font-weight: bolder;
    background-color: rgb(97, 50, 50);
    border-color: rgb(243, 222, 222);
    border-radius: 10px;
    height: 100%;
    width: 40%;
  }

  .required {
    color: var(--softRose);
  }

  .question {
    font-weight: bold;
  }

  @media (min-width: 900px) {
    .shipping {
      width: 400px;
    }
  }