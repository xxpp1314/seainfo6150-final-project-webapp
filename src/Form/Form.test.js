import React from "react";
import { render } from "@testing-library/react";
import Form from "./Form.jsx";

it("renders the Feedback component correctly", () => {
  const { container } = render(
    <Form />
  );
  expect(container).toMatchSnapshot();
});