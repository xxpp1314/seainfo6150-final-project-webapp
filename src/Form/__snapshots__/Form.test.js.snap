// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Feedback component correctly 1`] = `
<div>
  <div>
    <div>
      <div
        class="formdetails"
      >
        <form
          class="text"
        >
          <br />
          <label
            for="name"
          >
            Your First Name: 
          </label>
          <br />
          <input
            class="space"
            id="myFirstName"
            name="myFirstName"
            required=""
            type="text"
          />
          <br />
          <br />
          <label
            for="myTextId"
          >
            Your Last Name:  
          </label>
          <br />
          <input
            class="space"
            id="myLastName"
            name="myLastName"
            required=""
            type="text"
          />
          <br />
          <br />
          <label
            for="myTextId"
          >
            Your Email ID:  
          </label>
          <br />
          <input
            class="space"
            id="myEmailID"
            name="myEmailID"
            placeholder="abc@example.com"
            required=""
            type="email"
          />
          <br />
          <br />
          <label
            for="myTextId"
          >
            Your Phone No: 
          </label>
          <br />
          <input
            class="space"
            id="myPhone"
            name="myPhone"
            pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
            placeholder="123-345-6785"
            required=""
            type="tel"
          />
          <br />
          <br />
          <label
            for="myTextId"
          >
            Address Line 1: 
          </label>
          <input
            class="details"
            id="myAddressLine1"
            maxlength="500"
            name="myAddressLine1"
            required=""
            type="text"
          />
          <br />
          <br />
          <label
            for="myTextId"
          >
            Address Line 2: 
          </label>
          <input
            class="details"
            id="myAddressLine2"
            maxlength="500"
            name="myAddressLine2"
            type="text"
          />
          <br />
          <br />
          <label
            for="myTextId"
          >
            City of the Home Address: 
          </label>
          <br />
          <select
            class="dropdown"
            id="myCity"
            name="myCity"
          >
            <option
              value="Seattle"
            >
              Seattle
            </option>
            <option
              value="San Jose"
            >
              San Francisco
            </option>
            <option
              value="Boston"
            >
              Boston
            </option>
          </select>
          <br />
          <br />
          <label
            for="myTextId"
          >
            State of the Home Address: 
          </label>
          <br />
          <select
            class="dropdown"
            id="myState"
            name="myState"
          >
            <option
              value="WA"
            >
              Washington
            </option>
            <option
              value="CA"
            >
              California
            </option>
            <option
              value="MA"
            >
              Massachusetts
            </option>
          </select>
          <br />
          <br />
          <label
            for="myTextId"
          >
            Your Zip Code: 
          </label>
          <br />
          <input
            class="space"
            id="myZipCode"
            name="myZipCode"
            pattern="[0-9]{5}"
            placeholder="12345"
            required=""
            type="zip"
          />
          <br />
          <br />
          <label
            for="myTextId"
          >
            Do you have any experiences with pet?  
          </label>
          <select
            class="dropdown"
            id="myExperience"
            name="myExperience"
          >
            <option
              value="Yes"
            >
              Yes
            </option>
            <option
              value="No"
            >
              No
            </option>
          </select>
          <br />
          <br />
          <label
            for="myTextId"
          >
            What is your home type?  
          </label>
          <br />
          <select
            class="dropdown"
            id="myHome"
            name="myHome"
          >
            <option
              value="Apartment"
            >
              Apartment
            </option>
            <option
              value="House"
            >
              Single Family House/Town House
            </option>
            <option
              value="Other"
            >
              Other
            </option>
          </select>
          <br />
          <br />
          <label
            for="myTextId"
          >
            Does your home have any restrctions related to this pet? 
          </label>
          <br />
          <select
            class="dropdown"
            id="myRestriction"
            name="myRestriction"
          >
            <option
              value="Yes"
            >
              Yes
            </option>
            <option
              value="No"
            >
              No
            </option>
          </select>
          <br />
          <br />
          <input
            class="button"
            type="submit"
            value="Submit"
          />
          <br />
          <br />
        </form>
      </div>
    </div>
  </div>
</div>
`;
